src/procedures/proc_procedures.c

```diff
@@ -0,0 +1,89 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "proc_procedures.h"
+#include "../util/arr.h"
+#include "rax.h"
+#include "procedure.h"
+
+// CALL dbms.procedures()
+
+typedef struct {
+	raxIterator *current_proc; // Current proc
+	SIValue *output; // Array with a maximum of 4 entries: ["name", name, "mode", mode].
+} ProceduresContext;
+
+ProcedureResult Proc_ProceduresInvoke(ProcedureCtx *ctx,
+  const SIValue *args, const char **yield) {
+	if(array_len((SIValue *)args) != 0) return PROCEDURE_ERR;
+
+	ProceduresContext *pdata = rm_malloc(sizeof(ProceduresContext));
+
+	// Initialize an iterator to the rax that contains all procedures
+	rax* rax = Proc_Get_All();
+	raxIterator *it = rm_malloc(sizeof(raxIterator));
+	raxStart(it, rax);
+	raxSeek(it, "^", NULL, 0);
+	pdata->current_proc = it;
+	pdata->output = array_new(SIValue, 4);
+	pdata->output = array_append(pdata->output, SI_ConstStringVal("name"));
+	pdata->output = array_append(pdata->output, SI_ConstStringVal("")); // Place holder.
+	pdata->output = array_append(pdata->output, SI_ConstStringVal("mode"));
```

Comment:
maybe rename to "R/W"

Reply:
In neo4j they call it "mode", however they provide more options than R/W (e.g., "SCHEMA", "DBMS"). still thing we should change it?